packages/router-core/tests/searchParams.test.ts

```diff
@@ -2,43 +2,44 @@ import { describe, expect, test } from 'vitest'
 import { defaultParseSearch, defaultStringifySearch } from '../src'
 
 describe('Search Params serialization and deserialization', () => {
-  /*
-   * JSON-compatible objects can be serialized into a string,
-   * and then deserialized back into the original object.
-   */
   test.each([
-    [{}, ''],
-    [{ foo: '' }, '?foo='],
-    [{ foo: 'bar' }, '?foo=bar'],
-    [{ foo: 'bar baz' }, '?foo=bar+baz'],
-    [{ foo: 123 }, '?foo=123'],
-    [{ foo: '123' }, '?foo=%22123%22'],
-    [{ foo: true }, '?foo=true'],
-    [{ foo: 'true' }, '?foo=%22true%22'],
-    [{ foo: null }, '?foo=null'],
-    [{ foo: 'null' }, '?foo=%22null%22'],
-    [{ foo: 'undefined' }, '?foo=undefined'],
-    [{ foo: {} }, '?foo=%7B%7D'],
-    [{ foo: '{}' }, '?foo=%22%7B%7D%22'],
-    [{ foo: [] }, '?foo=%5B%5D'],
-    [{ foo: '[]' }, '?foo=%22%5B%5D%22'],
-    [{ foo: [1, 2, 3] }, '?foo=%5B1%2C2%2C3%5D'],
-    [{ foo: '1,2,3' }, '?foo=1%2C2%2C3'],
-    [{ foo: { bar: 'baz' } }, '?foo=%7B%22bar%22%3A%22baz%22%7D'],
-    [{ 0: 1 }, '?0=1'],
-    [{ 'foo=bar': 1 }, '?foo%3Dbar=1'],
-    [{ '{}': 1 }, '?%7B%7D=1'],
-    [{ '': 1 }, '?=1'],
-    [{ '=': '=' }, '?%3D=%3D'],
-    [{ '=': '', '': '=' }, '?%3D=&=%3D'],
-    [{ 'foo=2&bar': 3 }, '?foo%3D2%26bar=3'],
-    [{ 'foo?': 1 }, '?foo%3F=1'],
-    [{ foo: 'bar=' }, '?foo=bar%3D'],
-    [{ foo: '2&bar=3' }, '?foo=2%26bar%3D3'],
```

Comment:
Would it be possible to put those back the other way around? It might be a bit of a nitpick, but having these flipped
```diff
- [{}, ''],
+ ['', {}],
```
makes it harder to see what changed and what didn't. And for a PR like this, we're really trying to review this stuff character by character.